### ISSUE TITLE
恢复训练的代码中，如果不是分布式训练（train_sampler 为None），请问恢复时样本的顺序问题！

### ISSUE THREAD

**Issue by github-luffy** [jingyaogong/minimind#644]
根据代码逻辑，这里可能存在不一致性：

正常训练：有shuffle控制
shuffle=(train_sampler is None) 
恢复训练：如果train_sampler为None，使用range(len(train_ds))

这意味着：
正常训练时（没有sampler）：会shuffle
恢复训练时（没有sampler）：不会shuffle

如果要保持一致性，应该修改SkipBatchSampler的创建：
if train_sampler is None:
    # 创建一个随机采样器
    random_sampler = torch.utils.data.RandomSampler(train_ds)
    batch_sampler = SkipBatchSampler(random_sampler, args.batch_size, start_step + 1)
else:
    batch_sampler = SkipBatchSampler(train_sampler, args.batch_size, start_step + 1)

大佬这样的设计，可能的原因：

恢复训练是特殊情况：可能认为恢复训练时顺序不重要

简化逻辑：使用range(len(train_ds))最简单

分布式训练为主：代码可能主要针对分布式场景（总有train_sampler）。

**Comment by jingyaogong** [comment]
https://github.com/jingyaogong/minimind/commit/f7ffdf1fdb4fb4acb40f27d22150fb05bae7e283

2天前有过针对这个问题的更新 可以试试😊
